test/tests.py

```diff
@@ -85,24 +90,33 @@ def dmlc_reaper_args(exitcode_file, timeout_multiplier = 1):
 
 common_cflags = ["-O2", "-std=gnu99", '-Wall', '-Werror', '-Wpointer-arith',
                  '-Wwrite-strings', '-Wformat-nonliteral',]
+cc = os.environ.get('DMLC_CC')
 if is_windows():
-    cc = [testparams.mingw_cc()]
+    if not cc:
+        try:
+            cc = testparams.mingw_cc()
+            cc_not_found = not os.path.exists(cc)
+        except Exception:
+            cc_not_found = True
+        if cc_not_found:
+            raise TestFail('gcc not found(specify gcc by env var DMLC_CC)')
+        ldflags = [f"-L{testparams.mingw_root()}/lib/gcc/x86_64-w64-mingw32/lib"]
+    else:
+        ldflags = []
```

Comment:
Looks strange that the -L is needed in one branch but not the other, but if it works without the -L flag in your env, then I'm happy for now.

Reply:
I don't need -L in my env. Let's leave ldflags as [] for now. In long term, I may need to parameterize ldflags, as an env var,  for test purpose.